File: from-0601-to-0700/0645-set-mismatch/solution.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        doubled = None
        missed = None
                
        marks = [0] * len(nums)
        for num in nums:
            marks[num - 1] += 1
            if marks[num - 1] == 2:
                doubled = num
            
        for i, mark in enumerate(marks, 1):
            if mark == 0:
                missed = i
                break
                       
        return [doubled, missed]
        

File: from-0601-to-0700/0645-set-mismatch/solution.go
func findErrorNums(nums []int) []int {
    cnt := make(map[int]int)
    var twice, missing int
    for _, num := range nums {
        cnt[num] += 1
        if cnt[num] == 2 {
            twice = num
        }
    }
    for num := 1; num <= len(nums); num ++ {
        if cnt[num] == 0 {
            missing = num
            break
        }
    }
    return []int{twice, missing}